pkg/setting/setting_session_test.go

```diff
@@ -0,0 +1,43 @@
+package setting
+
+import (
+	"path/filepath"
+	"testing"
+
+	"github.com/grafana/grafana/pkg/log"
+	. "github.com/smartystreets/goconvey/convey"
+)
+
+type testLogger struct {
+	log.Logger
+	warnCalled  bool
+	warnMessage string
+}
+
+func (stub *testLogger) Warn(testMessage string, ctx ...interface{}) {
+	stub.warnCalled = true
+	stub.warnMessage = testMessage
+}
+func TestSessionSettings(t *testing.T) {
+	Convey("session config", t, func() {
+		skipStaticRootValidation = true
```

Comment:
Don't like that we have to mutate a global state in order to make the tests, strictly speaking I would have to revert that value, but since we are not doing it anywhere else...

To my mind it's very similar of the @bergquist concern here - https://github.com/grafana/grafana/pull/16445#discussion_r273353515

I guess settings module is overdue for refactoring?

Reply:
We are moving setting values into the `settings.Cfg` struct as we go. Rewriting all global vars at once feels like quite a big task right now since we need to pass the `Cfg` object around.